### DISCUSSION TITLE
Forbidden Error

### DISCUSSION THREAD

**Discussion by saltymedic** [netbox-community/netbox#11877] [Q&A]
I'm trying to install netbox on Ubuntu 22.04 using Apache. After I've gotten everything installed, when I try to visit the URL for my installation, I get a Forbidden error

`Forbidden
You don't have permission to access this resource.

`
It makes me think that something doesn't have correct ownership or permissions. Currently Netbox has ownership of root except for the media folder. From what I can tell, there's no place that says to change the ownership of other files in the Netbox documentation.

What am I missing?

Thanks

Jason

**Comment by candlerb** [comment]
First, try to identify whether this is a problem with Apache or a problem with Netbox.

If you run Netbox in [development mode](https://docs.netbox.dev/en/stable/installation/3-netbox/#test-the-application), does it work?

What exact URL are you looking at when you see the "Forbidden" message?  What does Apache's log files say (e.g. /var/log/apache/error.log)?  Does "dmesg" show any permission errors?

Are files in `/opt/netbox/netbox/static` *readable* to the Apache (www-data) user - including the enclosing directories?

Apart from the static files, Apache doesn't do anything except proxy the HTTP request to Netbox's gunicorn - so there should be no other permission issues around Apache, as long as it's not being constrained by SELinux or Apparmor (which "dmesg" would show)

**Comment by saltymedic** [comment]
Thank you for replying

I just tried reinstalling following the documentation. I must have missed something as it still isn't working.

I'm able to get apache to respond with its Apache2 Default Page. The problem comes when I try to get it to access Netbox.

When running Netbox in development mode, the output shows that it works:

`python3 manage.py runserver 0.0.0.0:8000 --insecure
Performing system checks...

System check identified no issues (0 silenced).
March 03, 2023 - 16:04:33
Django version 4.1.4, using settings 'netbox.settings'
Starting development server at http://0.0.0.0:8000/
Quit the server with CONTROL-C.
`

I'm installing this remotely, so I can't try the 0.0.0.0:8000 in a browser at this time.

The URL I am using is [https://netbox.risenlifeutah.org](https://netbox.risenlifeutah.org)

Within **/var/log/apache2/error.log** I have several lines that say:

`[Fri Mar 03 00:59:43.845084 2023] [authz_core:error] [pid 213080:tid 140052541716032] [client 71.6.146.185:54280] AH01630: client denied by server configuration: /opt/netbox/netbox/static`

Although there was nothing in the installation steps stating to change ownership of the static directory, I tried to see if that made a difference and it did not

`drwxr-xr-x 11 www-data www-data 4096 Mar  3 17:32 static/
`

Thanks 
Jason

**Reply by candlerb** [reply]
As the error says, your Apache config includes something that blocks accesses from your client IP address.  I'm afraid you'll have to go through your whole Apache config.

Netbox only provides the extra bits for a Netbox virtual server [here](https://github.com/netbox-community/netbox/blob/develop/contrib/apache.conf); it can't provide the entire config, and assumes you're starting from a "clean" config such as the default provided by Ubuntu.  Places to look include /etc/apache2/sites-enabled/ and /etc/apache2/conf-enabled/ on a typical system.

**Reply by saltymedic** [reply]
That gives me something to look into. Thank you!

Jason
